packages/blocks/src/class-blocks.php

```diff
@@ -17,7 +17,59 @@
  * @since 9.0.0
  */
 class Blocks {
+	/**
+	 * Get CSS classes for a block.
+	 *
+	 * @since 9.0.0
+	 *
+	 * @param string $slug  Block slug.
+	 * @param array  $attr  Block attributes.
+	 * @param array  $extra Potential extra classes you may want to provide.
+	 *
+	 * @return string $classes List of CSS classes for a block.
+	 */
+	public static function classes( $slug = '', $attr, $extra = array() ) {
+		if ( empty( $slug ) ) {
+			return '';
+		}
 
+		// Basic block name class.
+		$classes = array(
+			'wp-block-jetpack-' . $slug,
+		);
+
+		// Add alignment if provided.
+		if (
+			! empty( $attr['align'] )
+			&& in_array( $attr['align'], array( 'left', 'center', 'right', 'wide', 'full' ), true )
+		) {
+			array_push( $classes, 'align' . $attr['align'] );
+		}
+
+		// Add custom classes if provided in the block editor.
+		if ( ! empty( $attr['className'] ) ) {
+			array_push( $classes, $attr['className'] );
+		}
+
+		// Add any extra classes.
+		if ( is_array( $extra ) && ! empty( $extra ) ) {
+			$classes = array_merge( $classes, array_filter( $extra ) );
+		}
+
+		return implode( ' ', $classes );
+	}
+
+	/**
+	 * Does the page return AMP content.
+	 *
+	 * @return bool $is_amp_request Are we on am AMP view.
+	 */
+	public static function is_amp_request() {
+		$is_amp_request = ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() );
+
+		/** This filter is documented in 3rd-party/class.jetpack-amp-support.php */
+		return apply_filters( 'jetpack_is_amp_request', $is_amp_request );
```

Comment:
Would it be less code duplication to do like
```
return ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() );
```
instead? i.e. actually calling that instead of duplicating it, complete with reference to a filter that's defined there instead of here?

Reply:
I'm afraid that's not an option here. The package will be used by plugins other than Jetpack as well, so we can't really rely on the class.